features/step_definitions/user_steps.rb

```diff
@@ -1,18 +1,23 @@
 def create_visitor
+  @study_id_regexp_str = "\\AA[0-4]{1}[0-9]{1}[2-4]{1}[0-9]{4}\\z"
+  @study_id_regexp = Regexp.new(@study_id_regexp_str)
```

Comment:
@ammazzaw You were right about property-based testing! :smiling_face_with_tear:

Reply:
Haha yep 🤷‍♂️ But it's always good to see real life examples to confirm you need it 😂